utilities/merge_operators.h

```diff
@@ -20,6 +20,8 @@ class MergeOperators {
   static std::shared_ptr<MergeOperator> CreateUInt64AddOperator();
   static std::shared_ptr<MergeOperator> CreateStringAppendOperator();
   static std::shared_ptr<MergeOperator> CreateStringAppendOperator(char delim_char);
+  static std::shared_ptr<MergeOperator> CreateStringAppendOperator(
```

Comment:
Since this is a behavior change of StringAppend merge operator, also mention this change in HISTORY.md.

Reply:
@zhichao-cao This header is not in the public API.  Should it still be mentioned in HISTORY or not?

Reply:
@mrambacher 
> I generally agree with these changes (and made many of the same ones in #8481).
> 
> My one concern is in the stringappend_test. The way it is set up now it does not look as if the old 'char' interface is ever invoked, which would reduce the test coverage. Is it possible to somehow restructure the tests so that both the new and old constructors can be exercised?

I went for the simplest solution switching between the two constructors depending on the length of a delimiter specified in a test. I don't want to add another layer of indirection on top of `StringAppendOperatorTest::OpenFuncPtr` to accommodate two possible signatures.

Reply:
I suggest to mention it in the HISTORY.md, since this merge operator is used by many users and  they should be area of this behavior change. So mention it in the HISTORY.md would be better.

Reply:
I removed the mention of this specific header from HISTORY.md, leaving just `StringAppendOperator` itself.